### PR TITLE
Fix plans

### PR DESCRIPTION
**Pull request by gmuraru**
## Description
Plans changed such that the operations are not traced with the ```Tracer``` anymore. The ```PlaceHolder``` class is used to trace the operations.

## Type of change

Please mark the options that are relevant.

- [ ] Added/Modified tutorials
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

* [x] I did follow the [contribution guidelines](https://github.com/OpenMined/PySyft/blob/master/CONTRIBUTING.md)
* [x] I have commented my code following [Google style](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html).

### CODE REVIEW

File: syft/execution/placeholder.py [hunk 1]
```diff
@@ -95,19 +95,6 @@ def handle_func_command(cls, command):
 
         return response
 
-    def __getattribute__(self, name):
```
Comment (gmuraru) [inline syft/execution/placeholder.py:98]: Currently, we add the ```get_plain_text```to the ```PlaceHolder``` in the Torch Hook.

File: syft/frameworks/crypten/context.py [hunk 2]
```diff
@@ -102,9 +103,14 @@ def wrapper(*args, **kwargs):
             world_size = len(workers) + 1
             return_values = {rank: None for rank in range(world_size)}
 
+
+            hook_plan_building()
```
Comment (gmuraru) [inline syft/frameworks/crypten/context.py:107]: This is needed such that when we build the plan, we use another set of methods (currently only ```load``` is changed).

File: syft/frameworks/torch/hook/hook.py [hunk 3]
```diff
@@ -130,14 +131,16 @@ def __init__(
...
 
-        self.to_auto_overload = {}
+        self.to_auto_overload = defaultdict(list)
 
         self.args_hook_for_overloaded_attr = {}
 
         self._hook_native_tensor(torch.Tensor, TorchTensor)
 
         if dependency_check.crypten_available:
+            self.to_auto_overload[crypten.mpc.MPCTensor] = ["get_plain_text"]
```
Comment (gmuraru) [inline syft/frameworks/torch/hook/hook.py:141]: This will be also added to the PlaceHolder (such that we track this operation)

File: test/crypten/test_context.py [hunk 4]
```diff
@@ -17,7 +17,7 @@ def test_context(workers):
 
     @run_multiworkers([alice, bob], master_addr="127.0.0.1")
     @sy.func2plan()
-    def plan_func():
+    def plan_func(crypten=crypten):
```
Comment (youben11) [inline test/crypten/test_context.py:20]: Q: Why is this needed here?

Comment (gmuraru) [reply test/crypten/test_context.py:20]: Currently, there is a wrapped framework needed (and that will track the operations) -- in the ```syft/execution/tracing.py```

File: syft/execution/plan.py [hunk 5]
```diff
@@ -137,6 +137,10 @@ def __init__(
         if not hasattr(self, "forward"):
             self.forward = forward_func or None
 
+        self.wrapped_framework = {}
+        for f_name, f_packages in framework_packages.items():
+            self.wrapped_framework[f_name] = FrameworkWrapper(f_packages, self.role, self.owner)
+
```
Comment (LaRiffle) [inline syft/execution/plan.py:143]: Maybe a line of comment would be helpful to better understand the purpose of this

File: syft/execution/plan.py [hunk 6]
```diff
@@ -190,14 +194,15 @@ def build(self, *args):
...
             args += (self.state,)
 
-        with trace(framework_packages["torch"], self.role, self.owner) as wrapped_torch:
-            # Look for framework kwargs
-            framework_kwargs = {}
-            forward_args = inspect.getargspec(self.forward).args
-            if "torch" in forward_args:
-                framework_kwargs["torch"] = wrapped_torch
+        # Look for framework kwargs
+        framework_kwargs = {}
```
Comment (LaRiffle) [inline syft/execution/plan.py:198]: As well here: make sure to explain why we need to add frameworks kwargs

File: syft/frameworks/crypten/context.py (hunk 2 above)
Comment (LaRiffle) [reply syft/frameworks/crypten/context.py:106]: you can also put this as a comment in the code I'd say

File: syft/execution/plan.py (hunk 5 above)
Comment (gmuraru) [reply syft/execution/plan.py:143]: Done

File: syft/execution/plan.py (hunk 6 above)
Comment (gmuraru) [reply syft/execution/plan.py:198]: Done

File: syft/frameworks/crypten/context.py [hunk 7]
```diff
@@ -103,6 +103,10 @@ def wrapper(*args, **kwargs):
             world_size = len(workers) + 1
             return_values = {rank: None for rank in range(world_size)}
 
+            """
+            This is needed beause at building we use a set of methos written syft,
+             (currently only load is changed).
+            """
```
Comment (LaRiffle) [inline syft/frameworks/crypten/context.py:109]:
```suggestion
            # This is needed because at building we use a set of methods defined in syft (ex: load)
```
